hazelcast/src/main/java/com/hazelcast/map/impl/proxy/MapProxySupport.java

```diff
@@ -475,23 +486,42 @@ private MapOperation newPutTransientOperation(Data keyData, Data valueData,
                 timeInMsOrOneIfResultIsZero(maxIdle, maxIdleUnit));
     }
 
+    private InvocationBuilder overrideFailOnIndeterminateOperationState(InvocationBuilder invocationBuilder) {
+        SyncBackupTimeoutPolicy policy = getSyncBackupTimeoutPolicy();
```

Comment:
maybe we can delete `SyncBackupTimeoutPolicy` wholly and just have `boolean failOnIndeterminateOperationState`. It seems having it doesn't add much value.

Reply:
enum is better if we wanted to promote this to public API, because simple boolean would not be enough. But currently simple boolean will be sufficient for Jet.

Reply:
changed to boolean flag. `MapProxySupport` will completely ignore `failOnIndeterminateOperationState` value in `OperationService` for single-key operations